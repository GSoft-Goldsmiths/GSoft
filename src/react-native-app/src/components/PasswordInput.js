import React from 'react';
import { Image, Text, TextInput } from 'react-native';
import { color, fullPageView, mainFontBold } from '../styles/variables';

export default ({
                  value,
                  message,
                  placeholder,
                  handleTextChange,
                  handleSubmit,
                  imageSource,
                  title,
                  imageStyle,
                }) => {

  const Title = title
    ? <Text key="title" style={styles.title}>{title}</Text>
    : null;

  return [
    <Image
      key='image'
      style={{ ...fullPageView.image, ...imageStyle }}
      source={imageSource}
    />,
    Title,
    <TextInput
      key="textInput"
      ref={input => this._input = input}
      caretHidden
      autoFocus
      maxLength={4}
      keyboardType="numeric"
      style={styles.textInput}
      secureTextEntry={true}
      onChangeText={(text) => handleTextChange(text)}
      placeholder={placeholder}
      value={value}
      onSubmitEditing={(text) => handleSubmit(text)}
      onFocus={() => {console.log('It\'s focused now');}}
    />,
    <Text
      key='message'
      style={styles.message}>
      {message}
    </Text>,
  ];
}

const styles = {
  title: {
    ...fullPageView.title,
    fontSize: 20,
    fontFamily: mainFontBold,
    textAlign: 'center',
  },
  textInput: { height: 40, textAlign: 'center', fontSize: 20 },
  message: { color: color.danger, textAlign: 'center' },
};