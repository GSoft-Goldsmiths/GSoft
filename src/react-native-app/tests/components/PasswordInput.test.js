import React from 'react';
import PasswordInput from '../../src/components/PasswordInput';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create(<PasswordInput/>).toJSON();
  expect(rendered).toBeTruthy();
});
