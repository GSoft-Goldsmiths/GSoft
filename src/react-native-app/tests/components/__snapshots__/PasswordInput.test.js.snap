// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemListCell /> renders without crashing 1`] = `
Array [
  <Image
    source={
      Object {
        "uri": "source",
      }
    }
    style={
      Object {
        "height": 145,
        "marginBottom": 20,
        "width": 145,
      }
    }
  />,
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#0093A8",
        "fontFamily": "Caviar Dreams Bold",
        "fontSize": 20,
        "marginBottom": 15,
        "textAlign": "center",
      }
    }
  >
    page title
  </Text>,
  <TextInput
    allowFontScaling={true}
    autoFocus={true}
    caretHidden={true}
    keyboardType="numeric"
    maxLength={4}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="text input placeholder"
    secureTextEntry={true}
    style={
      Object {
        "fontSize": 20,
        "height": 40,
        "textAlign": "center",
      }
    }
    value="text input value"
  />,
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#C00",
        "textAlign": "center",
      }
    }
  >
    displayed message
  </Text>,
]
`;
